scripts/language_model/XLNet_classifier.py

```diff
@@ -0,0 +1,90 @@
+"""Model for sentence (pair) classification task/ regression with XLnet.
+"""
+from mxnet.gluon import Block
+from mxnet.gluon import nn
+import mxnet as mx
+
+
+class XLNetClassifier(Block):
+    """XLNet Classifier
+    """
+    def __init__(self, xl, units=768, num_classes=2, dropout=0.0,
+                 prefix=None, params=None):
+        super(XLNetClassifier, self).__init__(prefix=prefix, params=params)
+        self.xlnet = xl
+        self._units = units
+        with self.name_scope():
+            self.classifier = nn.HybridSequential(prefix=prefix)
+            if dropout:
+                self.classifier.add(nn.Dropout(rate=dropout))
+            self.classifier.add(nn.Dense(units=num_classes))
+            self.pooler = nn.Dense(units=units, flatten=False, activation='tanh', prefix=prefix)
+
+    def __call__(self, inputs, token_types, valid_length=None, mems=None):
+        # pylint: disable=arguments-differ
+        """Generate the unnormalized score for the given the input sequences.
+
+        Parameters
+        ----------
+        inputs : NDArray or Symbol, shape (batch_size, seq_length)
+            Input words for the sequences.
+        token_types : NDArray or Symbol, shape (batch_size, seq_length)
+            Token types for the sequences, used to indicate whether the word belongs to the
+            first sentence or the second one.
+        valid_length : NDArray or Symbol, or None, shape (batch_size)
+            Valid length of the sequence. This is used to mask the padded tokens.
+
+        Returns
+        -------
+        outputs : NDArray or Symbol
+            Shape (batch_size, num_classes)
+        """
+        return super(XLNetClassifier, self).__call__(inputs, token_types, valid_length, mems)
+
+    def _apply_pooling(self, sequence):
+        """Generate the representation given the inputs.
+
+        This is used for pre-training or fine-tuning a BERT model.
+        """
+        # for xlnet, we take the last hidden state
```

Comment:
Should this take the `valid_length` into account?

Reply:
Since we are using left pad, the CLS token will always at the end, so we may not need valid_length? I am going to add right pad, it will take valid_length then.